Lib/datetime.py

```diff
@@ -1244,6 +1335,18 @@ def isoformat(self, timespec='auto'):
 
     __str__ = isoformat
 
+    @classmethod
+    def fromisoformat(cls, time_string):
+        """Construct a time from the output of isoformat()."""
+        if not isinstance(time_string, str):
+            raise TypeError('fromisoformat: argument must be str')
```

Comment:
I'm not really sure why this is not getting hit by the test suite. `test_fromisoformat_fails_typeerror` is designed explicitly to hit this condition. Anyone have an idea why it's getting missed?